src/http/server.rs

```diff
@@ -63,20 +102,47 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
 
   new_headers
 }
+
+fn log_startup_messages(addr: &SocketAddr, enable_graphiql: bool) {
+  log::info!("🚀 Tailcall launched at [{}]", addr);
+  if enable_graphiql {
+    log::info!("🌍 Playground: http://{}", addr);
+  }
+}
+
 pub async fn start_server(config: Config) -> Result<()> {
   let blueprint = Blueprint::try_from(&config).map_err(CLIError::from)?;
   let http_client = Arc::new(DefaultHttpClient::new(&blueprint.upstream));
   let state = Arc::new(ServerContext::new(blueprint.clone(), http_client));
-  let make_svc = make_service_fn(move |_conn| {
+  let addr = (blueprint.server.hostname, blueprint.server.port).into();
+  let state_clone = state.clone();
+
+  let make_svc_single_req = make_service_fn(move |_conn| {
     let state = Arc::clone(&state);
-    async move { Ok::<_, anyhow::Error>(service_fn(move |req| handle_request(req, state.clone()))) }
+    async move { Ok::<_, anyhow::Error>(service_fn(move |req| handle_single_request(req, state.clone()))) }
   });
-  let addr = (blueprint.server.hostname, blueprint.server.port).into();
-  let server = hyper::Server::try_bind(&addr).map_err(CLIError::from)?.serve(make_svc);
-  log::info!("🚀 Tailcall launched at [{}]", addr);
-  if blueprint.server.enable_graphiql {
-    log::info!("🌍 Playground: http://{}", addr);
-  }
 
-  Ok(server.await.map_err(CLIError::from)?)
+  let make_svc_batch_req = make_service_fn(move |_conn| {
+    let state = Arc::clone(&state_clone);
+    async move { Ok::<_, anyhow::Error>(service_fn(move |req| handle_batch_request(req, state.clone()))) }
+  });
+
+  match blueprint.server.enable_batch_requests {
+    true => {
+      let server = hyper::Server::try_bind(&addr)
+        .map_err(CLIError::from)?
+        .serve(make_svc_batch_req);
+      log_startup_messages(&addr, blueprint.server.enable_graphiql);
+
+      Ok(server.await.map_err(CLIError::from)?)
+    }
+    false => {
+      let server = hyper::Server::try_bind(&addr)
+        .map_err(CLIError::from)?
+        .serve(make_svc_single_req);
+      log_startup_messages(&addr, blueprint.server.enable_graphiql);
+
+      Ok(server.await.map_err(CLIError::from)?)
+    }
+  }
```

Comment:
```suggestion

      let server = hyper::Server::try_bind(&addr)
        .map_err(CLIError::from)?
        .serve(if blueprint.server.enable_batch_requests { make_svc_batch_req } else { make_svc_single_req });
      log_startup_messages(&addr, blueprint.server.enable_graphiql);

      Ok(server.await.map_err(CLIError::from)?)
```

wdyt?

Reply:
This is the error I get 
```
`if` and `else` have incompatible types
expected struct `hyper::service::make::MakeServiceFn<[closure@src/http/server.rs:125:44: 125:56]>`
   found struct `hyper::service::make::MakeServiceFn<[closure@src/http/server.rs:120:45: 120:57]>`
no two closures, even if identical, have the same type
consider boxing your closure and/or using it as a trait object
```
I think if I try to Box them or make it a trait obj, it will insist on it being a `dyn`

Reply:
ofc...

at least we can strip it to some degree

```rust
  let builder = hyper::Server::try_bind(&addr).map_err(CLIError::from)?;
  log_startup_messages(&addr, blueprint.server.enable_graphiql);

  Ok(if blueprint.server.enable_batch_requests {
    builder.serve(make_svc_batch_req).await
  } else {
    builder.serve(make_svc_single_req).await
  }.map_err(CLIError::from)?)
  ```